slsim/Sources/SourceVariability/variability.py

```diff
@@ -36,4 +77,346 @@ def variability_at_time(self, observation_times):
         :param observation_times: image observation time
         :return: variability at given time.
         """
+
         return self._model(observation_times)
+
+    def sample_intrinsic_signal(self, observation_times):
+        """Interpolate to desired time step t, from a function defined by time_stamps
+        and values.
+
+        :param t: observation time in [day].
+        :param time_stamps: times associated with values in [day].
+        :param values: amplitude of signal at time_stamps [arbitrary units].
+        :return: variability at given time t
+        """
+        interpolation = scipy.interpolate.interp1d(
+            self._model[0],
+            self._model[1],
+            bounds_error=False,
+            fill_value=(self._model[1][0], self._model[1][-1]),
+        )
+        return interpolation(observation_times)
+
+
+class Accretion_Disk_Reprocessing(object):
```

Comment:
Class name should be AccretionDiskReprocessing(). This is a general convention. It would be great if you could create a separate python file for this class. We prefer to have a single class in a file.